### PR TITLE
Loadtree

### PR DESCRIPTION
**Pull request by albert-smit**
the Phylogenetic tree is now also loaded from a file, and opened from the open dialog

### CODE REVIEW

File: src/main/java/nl/tudelft/lifetiles/core/controller/MenuController.java [hunk 1]
```diff
@@ -132,6 +137,32 @@ private String collectGraphFileName(final File directory)
     }
 
     /**
+     * Find the file name of the newick file in the directory.
+     *
+     * @param dir
+     *            the directory in which the .nwk file is located
+     * @return the name of the nwk file without its extension
+     * @throws IOException
+     *             throws an exception when no, or too many files have been
```
Comment (jorenham) [inline src/main/java/nl/tudelft/lifetiles/core/controller/MenuController.java:159]: Iffy indentation

File: src/main/java/nl/tudelft/lifetiles/core/controller/ViewController.java [hunk 2]
```diff
@@ -169,6 +187,37 @@ public boolean isLoaded() {
...
+        File file;
+        Scanner sc = null;
+        try {
+            file = new File(fileName);
+            sc = new Scanner(file).useDelimiter("\\Z");
+        } catch (FileNotFoundException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+        String fileString = sc.next();
```
Comment (jorenham) [inline src/main/java/nl/tudelft/lifetiles/core/controller/ViewController.java]: Beware the potential NullPointer

File: src/main/java/nl/tudelft/lifetiles/core/controller/ViewController.java [hunk 3]
```diff
@@ -169,6 +187,37 @@ public boolean isLoaded() {
...
+        PhylogeneticTreeFactory np = new PhylogeneticTreeFactory(fileString);
```
Comment (jorenham) [inline src/main/java/nl/tudelft/lifetiles/core/controller/ViewController.java]: The factory ought to be passed to the method

**Comment by alangerak** [comment]
Looks good when above points are fixed

**Comment by rutgervandenberg** [comment]
- loadTree should take a File instead of a string.
- collectTreeFileName: assert is usually disabled, so this check should be made in a different way.
- PhylogeneticTreeFactory: this is not a factory.

**Comment by joswinter** [comment]
Looks good. I think it can be merged if you change the assert.

**Comment by rutgervandenberg** [comment]
Looks good.

File: src/main/java/nl/tudelft/lifetiles/core/controller/ViewController.java [hunk 4]
```diff
@@ -176,6 +193,37 @@ public boolean isLoaded() {
...
+        //convert the file to a single string
+        String fileString = null;
+        try (Scanner sc = new Scanner(treeFile).useDelimiter("\\Z")) {
+            fileString = sc.next();
+        } catch (FileNotFoundException e) {
+            throw e;
+        }
+
+        //parse the string into a tree
+        PhylogeneticTreeParser np = new PhylogeneticTreeParser(fileString);
```
Comment (jorenham) [inline src/main/java/nl/tudelft/lifetiles/core/controller/ViewController.java]: This is not very DI, you should use a factory or make it static.

**Pull request merged**